cmd/controller/root.go

```diff
@@ -201,6 +202,17 @@ var rootCmd = &cobra.Command{
 				os.Exit(1)
 			}
 		}
+		if err = (&generatorstate.Reconciler{
+			Client:     mgr.GetClient(),
+			Log:        ctrl.Log.WithName("controllers").WithName("GeneratorState"),
+			Scheme:     mgr.GetScheme(),
+			RestConfig: mgr.GetConfig(),
+		}).SetupWithManager(mgr, controller.Options{
+			MaxConcurrentReconciles: concurrent,
```

Comment:
Remember to rebase on https://github.com/external-secrets/external-secrets/pull/4339 and add `RateLimiter: ctrlcommon.BuildRateLimiter(),` here.